In [30]:
import pandas as pd

print("We beginning the history of our dataset")
print("Exploration")


We beginning the history of our dataset
Exploration


In [31]:
# télécharger le csv de l'hospitalisation suite à la COVOID le 31/03/2023
df = pd.read_csv("../data/covid-hospit-2023-03-31-18h01.csv", sep=";")

# voyons comment se présente notre po^pulation: aperçu de notre dataset
df_head = df.head()
print(df_head)

  dep  sexe        jour  hosp  rea  HospConv  SSR_USLD  autres  rad  dc
0  01     0  2020-03-18     2    0       NaN       NaN     NaN    1   0
1  01     1  2020-03-18     1    0       NaN       NaN     NaN    1   0
2  01     2  2020-03-18     1    0       NaN       NaN     NaN    0   0
3  02     0  2020-03-18    41   10       NaN       NaN     NaN   18  11
4  02     1  2020-03-18    19    4       NaN       NaN     NaN   11   6


In [32]:
# nombre de lignes et de colonnes de notre BDD
df_col_ligne = df.shape

print(f"ce dataset est constitué de: \n {df_col_ligne[0]} lignes et \n {df_col_ligne[1]} colonnes")

ce dataset est constitué de: 
 338245 lignes et 
 10 colonnes


In [25]:
# liste de colonnes
df_col = df.columns
print(list(df_col))

['dep', 'sexe', 'jour', 'hosp', 'rea', 'HospConv', 'SSR_USLD', 'autres', 'rad', 'dc']


In [ ]:
# afficher l'ensemble des informations sur notre dataset: 
# types de données, valeurs manquantes, nombres de lignes

df_info = df.info()

print(df_info)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 338245 entries, 0 to 338244
Data columns (total 10 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   dep       338245 non-null  object 
 1   sexe      338245 non-null  int64  
 2   jour      338245 non-null  object 
 3   hosp      338245 non-null  int64  
 4   rea       338245 non-null  int64  
 5   HospConv  228140 non-null  float64
 6   SSR_USLD  228140 non-null  float64
 7   autres    228140 non-null  float64
 8   rad       338245 non-null  int64  
 9   dc        338245 non-null  int64  
dtypes: float64(3), int64(5), object(2)
memory usage: 25.8+ MB
None


nous observons qu'il existe des valeurs manquantes dans certaines colonnes: HospConv, SSR_USLD, autres
nous avon 3 types de données object, float et int

In [26]:
# statistiques descriptives rapides
df.describe()

,sexe,hosp,rea,HospConv,SSR_USLD,autres,rad,dc
count,338245.000000,338245.000000,338245.000000,228140.000000,228140.000000,228140.000000,338245.000000,338245.000000
mean,0.996721,113.804733,13.363086,61.495625,35.983782,2.994867,2817.618895,534.741238
std,0.815822,166.746827,28.396802,83.826313,51.997225,5.827842,4173.613321,730.473523
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,0.000000,23.000000,1.000000,15.000000,6.000000,0.000000,500.000000,105.000000
50%,1.000000,58.000000,4.000000,34.000000,18.000000,1.000000,1404.000000,278.000000
75%,2.000000,134.000000,13.000000,73.000000,44.000000,3.000000,3338.000000,653.000000
max,2.000000,3281.000000,855.000000,1115.000000,565.000000,170.000000,48210.000000,6463.000000


In [28]:
# vérifie les valeurs manquantes
df.isnull().sum()

dep              0
sexe             0
jour             0
hosp             0
rea              0
HospConv    110105
SSR_USLD    110105
autres      110105
rad              0
dc               0
dtype: int64

In [45]:
# le nombre de départements représentés dans notre dataset
nbr_dep= df["dep"].nunique()
print(f" Les données de notre DataFrame couvrent {nbr_dep} départements Français.")

 Les données de notre DataFrame couvrent 102 départements Français.


Pourquoi 102 départements? 
Le jeu de données couvre 102 codes départementaux. Ce nombre est légèrement supérieur aux 101 départements français habituellement recensés, ce qui suggère la présence d'un code administratif spécifique ou d'une collectivité supplémentaire. Une analyse complémentaire des codes départementaux permettra d'identifier cette particularité.

In [ ]:
# aperçu des différents départements
print(sorted(df.dep.unique()))


['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '21', '22', '23', '24', '25', '26', '27', '28', '29', '2A', '2B', '30', '31', '32', '33', '34', '35', '36', '37', '38', '39', '40', '41', '42', '43', '44', '45', '46', '47', '48', '49', '50', '51', '52', '53', '54', '55', '56', '57', '58', '59', '60', '61', '62', '63', '64', '65', '66', '67', '68', '69', '70', '71', '72', '73', '74', '75', '76', '77', '78', '79', '80', '81', '82', '83', '84', '85', '86', '87', '88', '89', '90', '91', '92', '93', '94', '95', '971', '972', '973', '974', '976', '978']


On peut voir que les 102 sont dûs à la présence des départements 2A et 2B fesant référence à la Corse-du-Sud et la Haute-Corse respectivement.

In [58]:
#La période que couvre l'hospitalisation de notre dataFrame

print(df.jour.unique())
print(f"le 1er jour est: {df.jour.min()}")
print(f"le dernier est: {df.jour.max()}")


['2020-03-18' '2020-03-19' '2020-03-20' ... '2023-03-29' '2023-03-30'
 '2023-03-31']
le 1er jour est: 2020-03-18
le dernier est: 2023-03-31
